subprojects/core/src/main/java/org/gradle/cache/internal/GradleUserHomeCleanupService.java

```diff
@@ -27,6 +27,7 @@
 
     private static final long MAX_UNUSED_DAYS_FOR_RELEASES = 30;
     private static final long MAX_UNUSED_DAYS_FOR_SNAPSHOTS = 7;
+    private static final String DISABLE_CACHE_CLEANUP_PROPERTY = "org.gradle.disable.cache.cleanup";
```

Comment:
This property doesn't follow what I see as the convention, which is `org.gradle.<item>.<property>`. This should be `org.gradle.cache.cleanup.disabled` in my opinion.